werkzeug/routing.py

```diff
@@ -1141,7 +1141,12 @@ def bind(self, server_name, script_name=None, subdomain=None,
             subdomain = self.default_subdomain
         if script_name is None:
             script_name = '/'
-        server_name = _encode_idna(server_name)
+
+        try:
+            server_name = _encode_idna(server_name)
+        except:
+            server_name = None
```

Comment:
E.g. here the input is from the app developer: We should inform them that their input is wrong instead of doing something else.